asciimatics/utilities.py

```diff
@@ -48,3 +50,127 @@ class _DotDict(dict):
     __getattr__ = dict.get
     __setattr__ = dict.__setitem__
     __delattr__ = dict.__delitem__
+
+
+class BoxTool(object):
+    """Tool for building boxes out of characters. Supports a variety of line
+    styles from `asciimatics.constants`:
+
+    * ``ASCII_LINE`` -- ASCII safe characters (0)
+    * ``SINGLE_LINE`` -- Unicode based single lined box (1)
+    * ``DOUBLE_LINE`` -- Unicode based double lined box (2)
+
+    Individual characters of a box can be accessed directly through attributes:
+
+    * ``up_left`` -- corner piece facing up and left
+    * ``up_right`` -- corner piece facing up and right
+    * ``down_left`` -- corner piece facing down and left
+    * ``down_right`` -- corner piece facing down and right
+    * ``h`` -- horizontal line 
+    * ``v`` --  vertical line 
+    * ``v_inside`` --  vertical line used inside the grid
+    * ``v_left`` -- vertical line with mid joiner facing to the left
+    * ``v_right`` -- vertical line with mid joiner facing to the right
+    * ``h_up`` -- horizontal line with a mid joiner facing up
+    * ``h_down`` -- horizontal line with a mid joiner facing down
+    * ``cross`` -- intersection between vertical and horizontal
+    """
+    def __init__(self, unicode_aware, style=SINGLE_LINE):
+        """**Initialization:**
+
+        :param unicode_aware: boolean indicating if the terminal is Unicode
+            aware. If False, will force the use of the ASCII style
+        :param style: line style specifier. Supports ``ASCII_LINE``,
+            ``SINGLE_LINE``, and ``DOUBLE_LINE``. Defaults to ``SINGLE_LINE``.
+        """
+        self.unicode_aware = unicode_aware
+        self.style = style
```

Comment:
Actually, I want to call the setter because the setter has the side effect of setting up all the down_right, etc values. 

I could see how this could be confusing. I'm adding a comment to make it clear that the current is the desired behaviour.

Reply:
Oh...  Of course!  Just a comment is fine.